src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -815,52 +830,68 @@ private static EntitySourceType GetEntitySourceType(string entityName, Entity en
-                    // The second fk would not be present if its a Many-One relationship.
+                    // Any of these may be present when it's a One-One relationship,
+                    // The second fk isn't present when it's a Many-One relationship.
                     // When the configuration file doesn't specify how to relate these entities,
-                    // at least 1 of the following foreign keys should be present.
+                    // at least 1 of the following foreign keys should be present (in the database metadata?).
 
                     // Adding this foreign key in the hopes of finding a foreign key
                     // in the underlying database object of the source entity referencing
                     // the target entity.
-                    // This foreign key may NOT exist for either of the following reasons:
-                    // a. this source entity is related to the target entity in an One-to-One relationship
+                    // This foreign key may NOT exist when:
+                    // a. this source entity is related to the target entity in a One-to-One relationship
                     // but the foreign key was added to the target entity's underlying source
                     // This is covered by the foreign key below.
                     // OR
                     // b. no foreign keys were defined at all.
                     AddForeignKeyForTargetEntity(
+                        sourceEntityName: entityName,
+                        relationshipName: relationshipName,
                         targetEntityName,
                         referencingDbTable: databaseTable,
                         referencedDbTable: targetDbTable,
                         referencingColumns: relationship.SourceFields,
                         referencedColumns: relationship.TargetFields,
+                        referencingEntityRole: RelationshipRole.Source,
+                        referencedEntityRole: RelationshipRole.Target,
                         relationshipData);
 
                     // Adds another foreign key definition with targetEntity.GetSourceName()
                     // as the referencingTableName - in the situation of a One-to-One relationship
                     // and the foreign key is defined in the source of targetEntity.
                     // This foreign key WILL NOT exist if its a Many-One relationship.
-                    AddForeignKeyForTargetEntity(
-                        targetEntityName,
-                        referencingDbTable: targetDbTable,
-                        referencedDbTable: databaseTable,
-                        referencingColumns: relationship.TargetFields,
-                        referencedColumns: relationship.SourceFields,
-                        relationshipData);
+                    // Skips this fk when target and source entities are the same (self-referencing).
```

Comment:
Adding a note about why the addition of foreign key is skipped (for self-referencing relationship) will be helpful when going through the logic in the future